node/core/backing/src/lib.rs

```diff
@@ -428,9 +450,27 @@ impl CandidateBackingJob {
 
 		let summary = self.table.import_statement(&self.table_context, stmt);
 
+		if let Some(ref summary) = summary {
+			if let Some(attested) = self.table.attested_candidate(
+				&summary.candidate,
+				&self.table_context,
+			) {
+				if self.backed.insert(summary.candidate) {
```

Comment:
If we do it this way we will miss some validity votes or even include a candidate that we got an invalid message for at a later point.

Reply:
If there is >= threshold votes and an `Invalid` message is received someone is lying so in any way this case has to be handled by misbehavior reporting?

Reply:
As far as I understand this function here: https://github.com/paritytech/polkadot/blob/bkchr-start-backing-fix/statement-table/src/generic.rs#L521

Its not a misbehavior to send an `Invalid` message and it would also be weird to make this a misbehavior.

Reply:
I always thought that the idea is this: issuing any statement that is not true (`Invalid` on a valid andidate and vice versa) is a misbehavior. For instance this can happen when candidates are re-checked by randomly assigned validators and in this case a dispute(?) has to happen

Reply:
> If we do it this way we will miss some validity votes or even include a candidate that we got an invalid message for at a later point.

I just removed the `Invalid` check for includability. But yeah, we might miss some votes. Maybe it makes more sense to have this structure:
  * Backing subsystem notifies `Provisioner` of the `CandidateReceipt` that is includable.
  * Provisioner then can fetch full `BackedCandidate` for any of those, which will include all votes on the candidate.

This gives us the benefits that the `Provisioner` (high-level code) gets to decide which backed candidates to include and we don't drop votes. To avoid round-trips on the overseer, the request to the `CandidateBacking` subsystem should be a `Vec<Hash>` with a `Vec<BackedCandidate>` response.